### ISSUE TITLE
symlinks result in duplicate package bundles

### ISSUE THREAD

**Issue by cherrydev** [monounity/karma-typescript#228]
This is an issue that will probably mostly affect people using monorepos, and it ended up producing very subtle bugs for me. Most of the time, this probably won't result in errors. In my case, the issue resulted in angular's `TestBed.configureTestingModule` inconsistently overriding providers and imports.

A simple example in a monorepo:

```
packageA/package.json
packageB/package.json
packageCommonDep/package.json
```

`packageA` and `packageB` both directly depend on `packageCommonDep`
`packageA` depends on `packageB`

lerna or yarn will result in the node_modules looking like this using symlinks:
```
                      ->symlink to packageB
packageA/node_modules/packageB/node_modules/packageCommonDep/
                      ->symlink to packageCommonDep
packageA/node_modules/packageCommonDep/
                      ->symlink to packageCommonDep
packageB/node_modules/packageCommonDep/
packageCommonDep/
```

### The upshot:
packageA and packageB are now using two identical but distinct copies of anything declared inside of `packageCommonDep`. This will cause subtle and maddening errors, for example `instanceof`, static members, or any place you compare functions or constructors to each other, or use a function or constructor as a Map key. This is how angular mostly resolves dependencies.

A quick test shows that resolving the filename with `fs.realPath` or `fs.realPathSync` during the `browserResolve` callbacks eliminates this problem. I propose we add an option to the `resolve:` configuration property that enables this behaviour.

**Comment by erikbarke** [comment]
Always using `fs.realpathSync` on the resolved filenames doesn't seem to affect bundling times much and all integration tests have passed, does the PR #234 solve your problem?

**Comment by cherrydev** [comment]
Yes, this is exactly what my proof-of-concept code does. If this really doesn't cause any regressions for anybody, it will work for me.

**Comment by erikbarke** [comment]
Fixed in #234, closing.
